src/backend/expression/case_expression.h

```diff
@@ -9,69 +9,74 @@
 // Copyright (c) 2015-16, Carnegie Mellon University Database Group
 //
 //===----------------------------------------------------------------------===//
-
+#include <utility>
 #include "backend/expression/abstract_expression.h"
 #include "backend/expression/expression_util.h"
 
 namespace peloton {
 namespace expression {
 
+// the first part is condition, and the second part is expression
+typedef std::pair<AbstractExpression *, AbstractExpression *> WhenClause;
+
 class CaseExpression : public AbstractExpression {
  public:
-  CaseExpression(ValueType vt, const std::vector<AbstractExpression *> &clauses,
-                 AbstractExpression *default_result)
+  CaseExpression(ValueType vt, const std::vector<WhenClause *> &clauses,
+                 WhenClause *default_clause)
```

Comment:
With the modification to the typedef above, use `std::move` to move the vector parameter's contents.  Also pass in an r-value `unique_ptr&&` for the default clause.